lib/spack/spack/util/web.py

```diff
@@ -643,17 +647,15 @@ def find_versions_of_archive(
     # Any conflicting versions will be overwritten by the list_url links.
     versions = {}
     matched = set()
-    for url in archive_urls + sorted(links):
+    for url in sorted(links):
```

Comment:
Hi @trws, apologies but there's a conflict here after merging https://github.com/spack/spack/pull/27021. This can be resolved by using your changes but adding `url = convert_to_posix_path(url)` at the beginning of this loop (and you will want to do the same down at line 670 i.e. `for url in archive_urls:`).

Reply:
Ok, I think I got that resolved.  Here's hoping the tests bear that out. 🤞